packages/material-ui-system/src/getThemeValue.js

```diff
@@ -9,43 +9,48 @@ import sizing from './sizing';
 import spacing from './spacing';
 import typography from './typography';
 
+const filterPropsMapping = {
+  borders: borders.filterProps,
+  display: display.filterProps,
+  flexbox: flexbox.filterProps,
+  grid: grid.filterProps,
+  positions: positions.filterProps,
+  palette: palette.filterProps,
+  shadows: shadows.filterProps,
+  sizing: sizing.filterProps,
+  spacing: spacing.filterProps,
+  typography: typography.filterProps,
+};
+
+const styleFunctionMapping = {
+  borders,
+  display,
+  flexbox,
+  grid,
+  positions,
+  palette,
+  shadows,
+  sizing,
+  spacing,
+  typography,
+};
+
+export const propToStyleFunction = Object.keys(filterPropsMapping).reduce((acc, styleFnName) => {
+  filterPropsMapping[styleFnName].forEach(propName => {
+    acc[propName] = styleFunctionMapping[styleFnName];
+  });
+
+  return acc;
+}, {});
+
 function getThemeValue(prop, value, theme) {
   const inputProps = {
     [prop]: value,
     theme,
   };
 
-  if (borders.filterProps.indexOf(prop) !== -1) {
```

Comment:
Instead of invoking indexOf for each style function, now I am pre-generating mapping for each css/system prop to style function. If it exists we invoke it, if not we return the simple `{ prop: value }` object.